File: packages/go/kswitch/client_test.go
package kswitch

import (
	"context"
	"encoding/json"
	"net/http"
	"net/http/httptest"
	"testing"
)

func TestRegisterAgentRequest(t *testing.T) {
	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
		if r.Method != http.MethodPost {
			t.Fatalf("method = %s, want POST", r.Method)
		}
		if r.URL.Path != "/api/v1/agents/register" {
			t.Fatalf("path = %s, want /api/v1/agents/register", r.URL.Path)
		}
		if r.Header.Get("Authorization") != "Bearer test-token" {
			t.Fatalf("authorization header not set")
		}

		var body RegisterAgentRequest
		if err := json.NewDecoder(r.Body).Decode(&body); err != nil {
			t.Fatal(err)
		}
		if body.DisplayName != "customer-onboarding-v1" {
			t.Fatalf("display name = %s", body.DisplayName)
		}

		w.Header().Set("Content-Type", "application/json")
		_, _ = w.Write([]byte(`{"id":"agent-123"}`))
	}))
	defer server.Close()

	client, err := NewClient(server.URL, "test-token")
	if err != nil {
		t.Fatal(err)
	}

	agent, err := client.Governance.RegisterAgent(context.Background(), &RegisterAgentRequest{
		DisplayName:    "customer-onboarding-v1",
		RecordType:     "AGENT",
		RiskTier:       "tier_2",
		OwningDivision: "Retail Banking",
		OwningTeam:     "onboarding-platform",
	})
	if err != nil {
		t.Fatal(err)
	}
	if agent["id"] != "agent-123" {
		t.Fatalf("agent id = %v", agent["id"])
	}
}

func TestAuditEventsQuery(t *testing.T) {
	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
		if r.URL.RawQuery != "agent_id=agent-123&event_type=shadow_denied&limit=25" {
			t.Fatalf("query = %s", r.URL.RawQuery)
		}
		_, _ = w.Write([]byte(`{"events":[]}`))
	}))
	defer server.Close()

	client, err := NewClient(server.URL, "test-token")
	if err != nil {
		t.Fatal(err)
	}

	if _, err := client.Audit.Events(context.Background(), "agent-123", "shadow_denied", 25); err != nil {
		t.Fatal(err)
	}
}

File: packages/typescript/test/client.test.ts
import assert from "node:assert/strict";
import { test } from "node:test";
import { KSwitchClient } from "../src/index.js";

test("registerAgent sends the expected request", async () => {
  const calls: Array<{ url: string; init?: RequestInit }> = [];
  const fetchImpl: typeof fetch = async (input, init) => {
    calls.push({ url: String(input), init });
    return new Response(JSON.stringify({ id: "agent-123" }), {
      status: 200,
      headers: { "content-type": "application/json" },
    });
  };

  const client = new KSwitchClient({
    baseUrl: "https://api.example.test",
    apiKey: "test-token",
    fetchImpl,
  });

  const result = await client.governance.registerAgent({
    display_name: "customer-onboarding-v1",
    record_type: "AGENT",
    risk_tier: "tier_2",
    owning_division: "Retail Banking",
    owning_team: "onboarding-platform",
  });

  assert.deepEqual(result, { id: "agent-123" });
  assert.equal(calls[0].url, "https://api.example.test/api/v1/agents/register");
  assert.equal(calls[0].init?.method, "POST");
  assert.equal((calls[0].init?.headers as Record<string, string>).authorization, "Bearer test-token");
  assert.equal(JSON.parse(String(calls[0].init?.body)).display_name, "customer-onboarding-v1");
});

test("audit events adds filters as query parameters", async () => {
  let requestedUrl = "";
  const fetchImpl: typeof fetch = async (input) => {
    requestedUrl = String(input);
    return new Response(JSON.stringify({ events: [] }), { status: 200 });
  };

  const client = new KSwitchClient({
    baseUrl: "https://api.example.test",
    apiKey: "test-token",
    fetchImpl,
  });

  await client.audit.events({ agent_id: "agent-123", event_type: "shadow_denied", limit: 25 });

  assert.equal(
    requestedUrl,
    "https://api.example.test/api/v1/audit/events?agent_id=agent-123&event_type=shadow_denied&limit=25",
  );
});

